src/opera/parser/tosca/csar.py

```diff
@@ -1,19 +1,92 @@
+import shutil
+import yaml
+
 from pathlib import Path
 from tempfile import TemporaryDirectory
 from zipfile import ZipFile
 
-import yaml
 from opera.error import ParseError
 
 
 class CloudServiceArchive:
-    def __init__(self, csar_name, csar_folder_path):
+    def __init__(self, csar_name, csar_folder_path=None):
         self._csar_name = csar_name
         self._csar_folder_path = csar_folder_path
         self._tosca_meta = None
         self._root_yaml_template = None
         self._metadata = None
 
+    def package_csar(self, output, service_template=None, csar_format="zip"):
+        try:
+            root_yaml_files = []
+            root_yaml_files.extend(Path(self._csar_name).glob('*.yaml'))
+            root_yaml_files.extend(Path(self._csar_name).glob('*.yml'))
+
+            if len(root_yaml_files) != 1:
+                raise ParseError("There should be one root level yaml "
+                                 "file in the root of the CSAR: {}."
+                                 .format(list(map(str, root_yaml_files))),
+                                 self)
+
+            if not service_template:
+                service_template = root_yaml_files[0].name
+            else:
+                if not Path(self._csar_name).joinpath(
+                        service_template).exists():
+                    raise ParseError('The supplied TOSCA service template '
+                                     'file "{}" does not exist in folder '
+                                     '"{}".'.format(service_template,
+                                                    self._csar_name), self)
+
+            meta_file_path = Path(
+                self._csar_name) / "TOSCA-Metadata" / "TOSCA.meta"
+            if meta_file_path.exists():
+                # check existing TOSCA.meta file
+                with meta_file_path.open() as meta_file:
+                    self._tosca_meta = yaml.safe_load(meta_file)
+
+                self._validate_csar_version()
+                self._validate_tosca_meta_file_version()
+                template_entry = self._get_entry_definitions()
+                self._get_created_by()
+                self._get_other_definitions()
```

Comment:
What does this do?

Reply:
This creates a portal to the underworld. Just kidding. This gets the entries `Created-By` and `Other-Definitions` from the TOSCA meta file. The two keynames are not required in the meta file but just optional. However, we don't use the anywhere right now but they will probably be used in the future when (and if) opera evolves. So this is actually not needed here (the same applies for the CSAR validation where these two lines are present as well) but we can keep this anyways to be prepared for further implementation. I'm also ok if you want me to remove these.

Reply:
I will remove those two (`_get_created_by` and `_get_other_definitions`) method calls because they do nothing. The same stands for `_get_author`. I've also added the comments that these functions are not called anywhere but are prepared for future uses.